### PR TITLE
Global sidebar: Add sidebar menu item click event

### PR DESCRIPTION
**Pull request by taipeicoder**
Related to https://github.com/Automattic/dotcom-forge/issues/5856

## Proposed Changes

This PR implements the menu item click event `calypso_global_sidebar_menu_item_click` for the global sidebar. The event has the following properties:

| Property | Description |
| --- | --- |
| section | sites \| site \| me \| read |
| path | The menu item link URL |

## Testing Instructions

* Head to `/sites` and ensure that the menu item click event is triggered.
* Head to the Global site view and ensure that the menu item click event is triggered.
* Head to `/me` and ensure that the menu item click event is triggered.
* Head to `/read` and ensure that the menu item click event is triggered.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/layout/global-sidebar/index.jsx
Comment (taipeicoder) [inline client/layout/global-sidebar/index.jsx:1]: This is just renaming index.jsx → main.jsx.

**Review by fushar — COMMENTED** [review]
When clicking menu items in the Global Site View sidebar, the event is fired as follows (for my site):

- section: `site`
- path: `/purchases/subscriptions/testfusharuntangling.wpcomstaging.com`

I think we could make the path to be something like `/purchases/subscriptions/:site`? This is what the `calypso_page_view` event does. What do you think?

**Comment by taipeicoder** [comment]
> When clicking menu items in the Global Site View sidebar, the event is fired as follows (for my site):
> 
> * section: `site`
> * path: `/purchases/subscriptions/testfusharuntangling.wpcomstaging.com`
> 
> I think we could make the path to be something like `/purchases/subscriptions/:site`? This is what the `calypso_page_view` event does. What do you think?

Good idea, I've updated the code 🙂 
I could also add the code in the Reader sidebar, but doesn't seem necessary for now since none of the current links contain site domain.

**Review by fushar — APPROVED** [review]
Tested 4 sections and they work as described 👍

**Pull request merged**